# Data Analysis Challenge (DAC) - Spring 2024
In this notebook, we will demonstrate how to read in datasets directly from Github and follow that with some explorations.

# Import the relevant packages

These are some packages you could consider using for your analysis based on what you plan to do.
* pandas - for data wrangling.
* Matplotlib and/or Seaborn - for static data visualization.
* Plotly - for intereactive data visualization
* numpy - for matrix computation.
* scipy - for matrix computation and linear algebra.
* statsmodels - for statistical analysis

In [ ]:
import pandas as pd
import seaborn as sns

In [ ]:
#Read in dataset directly from Github
df = pd.read_csv("https://raw.githubusercontent.com/cmich-stats-club/spring2024-dac/main/datasets/cbb.csv")

In [ ]:
#Check columns, data type and non-null values in the original dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2455 entries, 0 to 2454
Data columns (total 24 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   TEAM        2455 non-null   object 
 1   CONF        2455 non-null   object 
 2   G           2455 non-null   int64  
 3   W           2455 non-null   int64  
 4   ADJOE       2455 non-null   float64
 5   ADJDE       2455 non-null   float64
 6   BARTHAG     2455 non-null   float64
 7   EFG_O       2455 non-null   float64
 8   EFG_D       2455 non-null   float64
 9   TOR         2455 non-null   float64
 10  TORD        2455 non-null   float64
 11  ORB         2455 non-null   float64
 12  DRB         2455 non-null   float64
 13  FTR         2455 non-null   float64
 14  FTRD        2455 non-null   float64
 15  2P_O        2455 non-null   float64
 16  2P_D        2455 non-null   float64
 17  3P_O        2455 non-null   float64
 18  3P_D        2455 non-null   float64
 19  ADJ_T       2455 non-null  

There are 2,455 records across 24 columns in the dataset. The only columns with missing records are POSTSEASON and SEED. Are these missing on purpose or it is a data anomaly?

In [ ]:
#Set option to display all columns
pd.set_option('display.max_columns', None)

In [ ]:
#Display first 5 records
df.head()

TEAM CONF   G   W  ADJOE  ADJDE  BARTHAG  EFG_O  EFG_D   TOR  \
0  North Carolina  ACC  40  33  123.3   94.9   0.9531   52.6   48.1  15.4   
1       Wisconsin  B10  40  36  129.1   93.6   0.9758   54.8   47.7  12.4   
2        Michigan  B10  40  33  114.4   90.4   0.9375   53.9   47.7  14.0   
3      Texas Tech  B12  38  31  115.2   85.2   0.9696   53.5   43.0  17.7   
4         Gonzaga  WCC  39  37  117.8   86.3   0.9728   56.6   41.1  16.2   

   TORD   ORB   DRB   FTR  FTRD  2P_O  2P_D  3P_O  3P_D  ADJ_T   WAB  \
0  18.2  40.7  30.0  32.3  30.4  53.9  44.6  32.7  36.2   71.7   8.6   
1  15.8  32.1  23.7  36.2  22.4  54.8  44.7  36.5  37.5   59.3  11.3   
2  19.5  25.5  24.9  30.7  30.0  54.7  46.8  35.2  33.2   65.9   6.9   
3  22.8  27.4  28.7  32.9  36.6  52.8  41.9  36.5  29.7   67.5   7.0   
4  17.1  30.0  26.2  39.0  26.9  56.3  40.0  38.2  29.0   71.5   7.7   

  POSTSEASON  SEED  YEAR  
0        2ND   1.0  2016  
1        2ND   1.0  2015  
2        2ND   3.0  2018  
3        2ND   3.0  2019  
4        2ND   1.0  2017

In [ ]:
#Descriptive statistics for numeric attributes
df.describe()

G            W        ADJOE        ADJDE      BARTHAG  \
count  2455.000000  2455.000000  2455.000000  2455.000000  2455.000000   
mean     31.492464    16.284318   103.304481   103.304603     0.493957   
std       2.657401     6.610960     7.376981     6.605318     0.256244   
min      15.000000     0.000000    76.600000    84.000000     0.005000   
25%      30.000000    11.000000    98.300000    98.500000     0.282200   
50%      31.000000    16.000000   103.000000   103.500000     0.475000   
75%      33.000000    21.000000   108.000000   107.900000     0.712200   
max      40.000000    38.000000   129.100000   124.000000     0.984200   

             EFG_O        EFG_D          TOR         TORD          ORB  \
count  2455.000000  2455.000000  2455.000000  2455.000000  2455.000000   
mean     49.805132    50.001385    18.763055    18.689572    29.875642   
std       3.143061     2.939602     2.090595     2.201749     4.134332   
min      39.200000    39.600000    11.900000    10.200000    15.000000   
25%      47.750000    48.000000    17.300000    17.200000    27.100000   
50%      49.700000    50.000000    18.700000    18.600000    29.900000   
75%      51.900000    52.000000    20.100000    20.100000    32.600000   
max      59.800000    59.500000    27.100000    28.500000    43.600000   

               DRB          FTR        FTRD         2P_O         2P_D  \
count  2455.000000  2455.000000  2455.00000  2455.000000  2455.000000   
mean     30.075479    35.989776    36.26998    48.802281    48.976660   
std       3.151454     5.247820     6.24590     3.384468     3.340546   
min      18.400000    21.600000    21.80000    37.700000    37.700000   
25%      27.900000    32.400000    31.90000    46.500000    46.700000   
50%      30.000000    35.800000    35.80000    48.700000    49.000000   
75%      32.200000    39.500000    40.20000    51.000000    51.300000   
max      40.400000    58.600000    60.70000    62.600000    61.200000   

              3P_O         3P_D        ADJ_T          WAB        SEED  \
count  2455.000000  2455.000000  2455.000000  2455.000000  476.000000   
mean     34.406395    34.598737    67.812301    -7.802485    8.802521   
std       2.789434     2.415766     3.277622     6.965736    4.676354   
min      24.900000    27.100000    57.200000   -25.200000    1.000000   
25%      32.500000    33.000000    65.700000   -13.000000    5.000000   
50%      34.400000    34.600000    67.800000    -8.300000    9.000000   
75%      36.300000    36.200000    70.000000    -3.150000   13.000000   
max      44.100000    43.100000    83.400000    13.100000   16.000000   

              YEAR  
count  2455.000000  
mean   2016.007332  
std       1.999375  
min    2013.000000  
25%    2014.000000  
50%    2016.000000  
75%    2018.000000  
max    2019.000000

From the YEAR summaries, we see that this data is the seasons 2013 through 2019.

In [ ]:
#Descriptive statistics for categorical attributes
df.describe(include="object")

TEAM  CONF POSTSEASON
count             2455  2455        476
unique             355    35          8
top     North Carolina   ACC        R64
freq                 7   102        224

There are 355 teams (schools) spanning 35 conferences.

In [ ]:
#I want to focus on only the teams that made it to the post season
df_post_season = df[~df.POSTSEASON.isnull()]

In [ ]:
#Number of records in the post season subset table
df_post_season.shape

(476, 24)

In [ ]:
#Also get the no post season records
df_post_season_missed = df[df.POSTSEASON.isnull()]

In [ ]:
df_post_season_missed.shape

(1979, 24)

In [ ]:
#Which conferences had the most representation 2013-2019
df_post_season.CONF.value_counts()

ACC     48
B10     46
B12     45
BE      40
SEC     34
P12     32
A10     25
Amer    19
MWC     16
WCC     12
MVC     10
BSth     8
OVC      8
SB       8
BSky     7
SWAC     7
Pat      7
NEC      7
MEAC     7
MAAC     7
Horz     7
CAA      7
Slnd     7
Sum      7
AE       7
SC       7
MAC      7
Ivy      7
CUSA     7
BW       7
WAC      7
ASun     6
Name: CONF, dtype: int64

The ACC is the most represented conference, followed by the Big 10 (B10).   
CMU is in the MAC. This conference has one representative per season. Who are these MAC representatives in the March Madness?

In [ ]:
df_post_season[df_post_season.CONF=="MAC"].sort_values("YEAR", ascending=False)

TEAM CONF   G   W  ADJOE  ADJDE  BARTHAG  EFG_O  EFG_D  \
2091           Buffalo  MAC  35  32  113.6   95.3   0.8819   53.4   47.3   
2090           Buffalo  MAC  35  27  114.5  102.7   0.7761   54.8   48.8   
2245          Kent St.  MAC  35  22  104.1  102.9   0.5322   48.2   50.1   
2244           Buffalo  MAC  34  19  105.1  102.4   0.5746   49.6   49.8   
2243           Buffalo  MAC  33  23  109.1   98.0   0.7742   48.7   48.5   
2330  Western Michigan  MAC  32  22  104.7  102.5   0.5605   52.0   47.6   
2357             Akron  MAC  32  25  103.4   94.4   0.7402   51.0   44.6   

       TOR  TORD   ORB   DRB   FTR  FTRD  2P_O  2P_D  3P_O  3P_D  ADJ_T  WAB  \
2091  16.0  20.3  31.7  26.9  32.7  31.9  55.7  49.3  33.7  29.3   74.9  5.5   
2090  16.5  18.1  31.9  29.1  29.9  39.5  54.3  48.4  37.1  33.1   73.8 -1.3   
2245  17.9  19.0  38.1  29.8  32.5  34.3  48.5  48.5  31.7  35.3   69.5 -7.2   
2244  18.8  17.7  31.8  28.7  41.6  37.8  49.0  48.8  33.7  34.3   72.9 -5.8   
2243  16.7  19.7  34.5  30.4  42.6  34.3  47.6  48.0  34.1  33.0   68.9 -1.3   
2330  20.9  18.5  31.0  32.0  51.4  34.7  53.7  47.8  32.6  31.5   67.0 -2.9   
2357  20.8  20.2  37.6  30.5  36.9  35.0  52.1  42.7  32.7  32.8   66.5  0.3   

     POSTSEASON  SEED  YEAR  
2091        R32   6.0  2019  
2090        R32  13.0  2018  
2245        R64  14.0  2017  
2244        R64  14.0  2016  
2243        R64  12.0  2015  
2330        R64  14.0  2014  
2357        R64  12.0  2013

Buffalo is the most dominant team in the MAC, 4 March Madness appearances in 7 years. CMU need to improve their odds of making it to the NCAA tournament.  
Unfornutely, the farthest the MAC teams have gone is the Round of 32.